### ISSUE TITLE
Add user settings page

### ISSUE THREAD

**Issue by cloudrac3r** [cloudrac3r/bibliogram#28]
Right now, this is just to collect issues that need a user settings page before they can be implemented. \\o/

**Comment by cloudrac3r** [comment]
I will actually work on this soon, since there are so many issues that are waiting on it.

I would prefer if other people didn't contribute towards this issue without consulting me first, since I'd like be in control of request routing as well as the appearance of the settings page.

**Comment by mendel5** [comment]
You @cloudrac3r  are in full control :wink:

**Comment by cloudrac3r** [comment]
If you'd like to discuss the visual style of the controls on the settings page (there may also be custom themes that override these, but I don't know yet) you may find it interesting to read this blog post by me. If you only care about Bibliogram, scroll to the bottom. https://cadence.moe/blog/2020-04-28-settings-through-the-ages

**Comment by cloudrac3r** [comment]
[The settings page now exists,](https://bibliogram.art/settings) but none of the settings are implemented yet.

[This image shows what the page will look like when some controls are actually available.](https://cadence.moe/i/05c9ab)

I'm now realising that I don't have a consistent site header that I can link to the settings page in (unlike Invidious, for example). Fortunately, both the user page and the post page have a bit of extra space in the sidebars that I may be able to steal and use to hold a link to settings. I'm not so sure how to do it in the mobile view, though. Maybe just mobile will get a header.

**Comment by mendel5** [comment]
Thank you, looks very good!

**Comment by allendema** [comment]
When this will be impleted, it will save user settings in a cookie, right?
Since my browsers is set to delete all the cookies after quiting the app, it means that one has to change everytime the desired settings.
Ideally it would be the URL + parameter(s),
dark=1&theme=2 and so on.
Maybe you can take a look at [https://duckduckgo.com/settings#general](https://duckduckgo.com/settings#general)

**Comment by cloudrac3r** [comment]
That's a really good point. Here is a thing that I thought of that would be easy for me to code and still allow you to save settings:

A URL like bibliogram.art/?settings=4deb5 which will load the token from the URL parameter into the cookie. Then you can bookmark this URL with query string, and use that at the start of your session to create the cookie, so you can browse the rest of Bibliogram for that session. However, this requires you to remember to click the bookmark, and it requires you to **still save cookies, and to not clear them midway through the session.**

Is this good enough for your use case?

Or you could whitelist Bibliogram in your browser — you trust it, right? :D

— Cadence

**Comment by allendema** [comment]
If what you said about coding side is doable, I definetly would prefer that option.
Not clearing cookies midway thru the session is not a problem either.
As for whitelisting the domain, could not find a way to do it on ESR Firefox on Android.
But it is possible on the desktop version. ;)

**Comment by cloudrac3r** [comment]
Yay!

There's no cookie save/load endpoint yet. Please open a new issue for that so that I don't forget.

**Comment by mendel5** [comment]
Thank you, @cloudrac3r !
